### PR TITLE
Restructure docs and a small refactoring

### PR DESCRIPTION
**Pull request by vitorbaptista**
Please check the commit messages

### CODE REVIEW

**Comment by roll** [comment]
I'll be able to take a look today on evening.

**Comment by roll** [comment]
Still wasn't able to review for now. There are new commits - is it VIP or ready for a review?

**Comment by vitorbaptista** [comment]
@roll It's ready for review, but as I'm working on this area, it might change a bit. Nothing that blocks you taking a look.

**Comment by roll** [comment]
@pwalsh
@jobarratt
@amercader
@vitorbaptista
Documentation plan for goodtables family has never been discussed with me (not sure about Adria) and it feels kind wierd. So probably it's too late for sharing ideas but still my few cents:

- My vision was that the best way to achieve all goals here (incl. ODI deliverables) is to create goodtables.io/docs system which could treat goodtables.io and goodtables-py as one product (think "Travis and Travis-Local")
- After some thinking I found the @vitorbaptista's idea for local support of goodtables.yml just amazing. It's super easy to implement and it will make the concept of "Travis and Travis-Local" true for goodtables.
- After having so coupled software we could write it up as one product. Consider gootables.io/docs section structure for example:
  - What is goodtables quick introduction
  - What is tabular data quality and how to achieve it
  - What are data quality errors and list of supported by goodtables (here link to Data Quality Spec)
  - Continous validation using goodtables.io
    - how to setup
    - goodtables.yml
    - etc
  - Local data validation with goodtables-py
    - how to validate your goodtables.io project locally
    - how to use it with CLI arguments 
    - how to use it with Python (here link to README.md for full techincal documentataion)
    - some tutorial and examples
    - etc
  - Embedding goodtables, integrators docs

In my eyes something like this will be just amazing in all aspects. It will be having an integrity of finished product. It could use some nice consolidated documentation capabilities like fancy design, perfect navigation, cross-linking etc. 

And writing and re-writing REDMEs on Github - I just not sure it really moves as closer to having good documentation in overall. I don't mind in high-level will it be new version or old version. But it still just a README on GitHub, low-level thing.

File: README.md [hunk 1]
```diff
@@ -43,263 +66,81 @@ report['tables'][0]['source'] # 'invalid.csv'
...
+| [custom-constraint](#custom-constraint) | Defines a constraint based on the values of other columns (e.g. `value * quantity == total`). |
 
 #### blacklisted-value
 
 Sometimes we have to check for some values we don't want to have in out dataset. It accepts following options:
 
-- `column (int/str)` - column number or name
-- `blacklist (list)` - list of non-valid values
+<dl>
+  <dt>column (int/str)</dt>
```
[the remark this answers is not in the document]
Comment (vitorbaptista) [reply README.md]: Agreed. I changed it to tables.

File: tox.ini [hunk 2]
```diff
@@ -28,3 +27,10 @@ commands=
     --cov-config tox.ini \
     --cov-report term-missing \
     {posargs}
+
+[testenv:lint]
```
[the remark this answers is not in the document]
Comment (vitorbaptista) [reply tox.ini]: I shouldn't have done this change in the middle of this PR, as it has nothing to do with the other changes. However, I do think this is the best approach, so it was easier to submit a PR to oki-py than rolling back it here. https://github.com/okfn/oki-py/pull/12

**Comment by vitorbaptista** [comment]
Thanks for the thorough review, @roll. I've added back the changelog, semver notice, and ODS install support.

I'm not sure what you mean with the change to RST. Are you talking about the docstrings?

As you noticed, most of the changes here are around improving the formatting and the wording. The content is basically the same. I envision this README to be used more as a reference for the validation options than actually a tutorial. For an introduction/tutorial, I agree with you on a separate page (see https://github.com/frictionlessdata/goodtables.io/issues/213).

**Pull request merged**